apps/server/src/shared/domain/entity/task.entity.spec.ts

```diff
@@ -24,723 +22,816 @@ describe('Task Entity', () => {
 
-			expect(spy).toHaveBeenCalled();
-			expect(result.graded).toEqual(5);
+			it('should be create correct status for isDraft', () => {
+				const { task, user } = setup();
 
-			spy.mockReset();
+				const status = task.createTeacherStatusForUser(user);
+
+				expect(status.isDraft).toEqual(true);
+			});
 		});
 
-		it('should call getMaxSubmissions and return the result as maxSubmissions property', () => {
-			const user = userFactory.build();
-			const task = taskFactory.build();
-			const spy = jest.spyOn(task, 'getMaxSubmissions').mockImplementation(() => 5);
+		describe('when task is a finished for the user', () => {
+			const setup = () => {
+				const user = userFactory.buildWithId();
+				const task = taskFactory.isPublished().buildWithId({ creator: user, finished: [user] });
+
+				return { task, user };
+			};
 
-			const result = task.createTeacherStatusForUser(user);
+			it('should be create correct status for isFinished', () => {
+				const { task, user } = setup();
 
-			expect(spy).toHaveBeenCalled();
-			expect(result.maxSubmissions).toEqual(5);
+				const status = task.createTeacherStatusForUser(user);
 
-			spy.mockReset();
+				expect(status.isFinished).toEqual(true);
+			});
 		});
 
-		it('should call isDraft and return the result as isDraft property', () => {
-			const user = userFactory.build();
-			const task = taskFactory.build();
-			const spy = jest.spyOn(task, 'isDraft').mockImplementation(() => true);
+		describe('when user is a submissionTeacher in parent that is a course', () => {
+			const setup = () => {
+				const user = userFactory.buildWithId();
+				const course = courseFactory.buildWithId({ substitutionTeachers: [user] });
+				const task = taskFactory.isPublished().buildWithId({ creator: user, course });
+
+				return { task, user };
+			};
 
-			const result = task.createTeacherStatusForUser(user);
+			it('should be create correct status for isSubstitutionTeacher', () => {
+				const { task, user } = setup();
 
-			expect(spy).toHaveBeenCalled();
-			expect(result.isDraft).toBe(true);
+				const status = task.createTeacherStatusForUser(user);
 
-			spy.mockReset();
+				expect(status.isSubstitutionTeacher).toEqual(true);
+			});
 		});
+	});
 
-		describe('when parent is a course', () => {
-			it('should return true if the user is part of it.', () => {
-				const user = userFactory.build();
-				const course = courseFactory.build();
-				course.substitutionTeachers.add(user);
-				const task = taskFactory.build({ course });
+	describe('createStudentStatusForUser is called', () => {
+		describe('when submissions are not populated', () => {
+			const setup = () => {
+				const user = userFactory.buildWithId();
+				const task = taskFactory.buildWithId({});
+				Object.assign(task, { submissions: undefined });
 
-				const result = task.createTeacherStatusForUser(user);
+				return { task, user };
+			};
 
-				expect(result.isSubstitutionTeacher).toBe(true);
+			it('should be throw an error', () => {
```

Comment:
```suggestion
			it('should throw an error', () => {
```
Hab die weiteren nicht mehr angemerkt. Suche bitte selber mal nach 'be' im File